src/statsmodels/formula.jl

```diff
@@ -553,7 +557,7 @@ function coefnames(mf::ModelFrame)
         push!(term_names, expandtermnames(names))
     end
 
-    reduce(vcat, term_names)
+    isempty(term_names) ? Vector{Compat.UTF8String}() : reduce(vcat, term_names)
```

Comment:
Because `Compat.String` is `ByteString` on 0.4, whereas `Compat.UTF8String` is `UTF8String` on 0.4 and `String` on 0.5. If I'm not mistaken, the only possible output type from `coefnames` is `Vector{Compat.UTF8String}`, which is why I did that here. I like your `reduce` suggestion, I'll implement that later today (unless someone wants to push that to this branch in the meantime).